src/Persistence/MsApp/MsappArchive.cs

```diff
@@ -207,6 +207,60 @@ public static IMsappArchive Open(string path, IServiceProvider serviceProvider)
 
     #region Methods
 
+    /// <summary>
+    /// Deserializes the entry with the given name into an object of type T.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="entryName"></param>
+    /// <param name="ensureRoundTrip"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentNullException"></exception>
+    /// <exception cref="InvalidOperationException"></exception>
+    /// <exception cref="PersistenceException"></exception>
+    public T Deserialize<T>(string entryName, bool ensureRoundTrip = true)
+    {
+        if (string.IsNullOrWhiteSpace(entryName))
+            throw new ArgumentNullException(nameof(entryName));
+
+        var entry = GetRequiredEntry(entryName);
+        if (!entry.FullName.EndsWith(YamlFileExtension, StringComparison.OrdinalIgnoreCase))
```

Comment:
are we allowing yaml deserialization only?

Reply:
We'll json deserialization too but I did want to do it in this PR because it requires bringing in Json deserialization properties and other methods